### PR TITLE
feat(ci): add smoke test for domain mutation

### PR DESCRIPTION
**Pull request by anshbansal**
Follow-up https://github.com/datahub-project/datahub/pull/6259 to add tests for domains

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: smoke-test/tests/cypress/cypress/integration/mutations/domains.js [hunk 1]
```diff
@@ -14,12 +15,55 @@ describe("add remove domain", () => {
...
+
+    it("delete a domain and ensure dangling reference is deleted on entities", () => {
+        cy.login();
+        cy.goToDomainList();
+        cy.get('[data-testid="' + domain_created_urn + '"]').within(() => {
+            cy.get(".ant-dropdown-trigger").click();
+        });
+        cy.clickOptionWithText("Delete");
+        cy.clickOptionWithText("Yes");
+        cy.waitTextVisible("Deleted Domain!")
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/cypress/integration/mutations/domains.js:63]: (same here)

File: smoke-test/tests/cypress/cypress/support/commands.js [hunk 2]
```diff
@@ -83,6 +83,12 @@ Cypress.Commands.add("goToUserList", () => {
   cy.waitTextVisible("Manage Users & Groups");
 })
 
+Cypress.Commands.add("goToStarSearchList", () => {
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/cypress/support/commands.js:86]: nice!

File: smoke-test/tests/cypress/cypress/support/commands.js [hunk 3]
```diff
@@ -143,6 +149,12 @@ Cypress.Commands.add('addTermToDataset', (urn, dataset_name, term) => {
   cy.contains(term);
 });
 
+Cypress.Commands.add("removeDomainFromDataset", (urn, dataset_name, domain_urn) => {
+  cy.goToDataset(urn, dataset_name);
+  cy.get('[class^="EntitySidebar__ContentContainer"] [href="/domain/' + domain_urn + '"] .anticon-close').click();
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/cypress/support/commands.js:154]: This is not the ideal way to reference these HTML objects. Typically we use data-testid="custom-tag-id" to identify components. This way is prone to breaking with small refactoring, which can end up increasing ship-time

File: smoke-test/tests/cypress/cypress/integration/mutations/domains.js [hunk 4]
```diff
@@ -14,12 +15,55 @@ describe("add remove domain", () => {
...
+        cy.clickOptionWithText(test_domain);
+        cy.waitTextVisible("Add assets")
+        cy.clickOptionWithText("Add assets")     
+        cy.get(".ant-modal-content").within(() => {
+            cy.get('[data-testid="search-input"]').click().type("jaffle")
+            cy.waitTextVisible("jaffle_shop")
+            cy.get(".ant-checkbox-input").first().click()
+            cy.get("#continueButton").click()
+        })
+        cy.waitTextVisible("Added assets to Domain!")
```
[the remark this answers is not in the document]
Comment (anshbansal) [reply smoke-test/tests/cypress/cypress/integration/mutations/domains.js:34]: Domain page does not get updated when assets are added to it. You have to refresh the page before the assets appear. If we refresh the page before assets are added the request might not complete. So this is currently the only way to do this test reliably until we change the domain page to actually refresh the UI when asset addition is complete

[the remark this answers is not in the document]
Comment (gabe-lyons) [reply smoke-test/tests/cypress/cypress/integration/mutations/domains.js:34]: @anshbansal as a temporary placeholder let's just add a wait then, e.g. 5 seconds

[the remark this answers is not in the document]
Comment (anshbansal) [reply smoke-test/tests/cypress/cypress/integration/mutations/domains.js:34]: Adding wait times will turn this into a flaky test. I had removed all the temp wait times to remove flakiness in these tests.

[the remark this answers is not in the document]
Comment (jjoyce0510) [reply smoke-test/tests/cypress/cypress/integration/mutations/domains.js:34]: It's okay. This is fine for now.

**Review by szalai1 — APPROVED** [review]
approving it without context, based on @jjoyce0510 's comment

**Pull request merged**